engine/crates/telemetry/src/span.rs

```diff
@@ -35,9 +35,28 @@ pub trait GqlRecorderSpanExt {
     fn record_gql_request(&self, attributes: GqlRequestAttributes<'_>);
     /// Record GraphQL response attributes in the span
     fn record_gql_response(&self, attributes: GqlResponseAttributes);
+    /// Record subgraph response attributes in the span
+    fn record_subgraph_response(&self, attributes: SubgraphResponseAttributes);
+
+    fn record_gql_error(&self, error: String);
+    fn record_gql_duration(&self, duration: Duration);
+
+    fn record_gql_status(&self, status: GraphqlResponseStatus, duration: Duration, error: Option<String>) {
```

Comment:
You shouldn't add any errors in the traces yet. It's unclear how we'll process them and we'll certainly want more than just the first error message.